dask/dataframe/shuffle.py

```diff
@@ -981,12 +982,28 @@ def compute_and_set_divisions(df, **kwargs):
         or any(a > b for a, b in zip(mins, maxes))
     ):
         raise ValueError(
-            "Partitions must be sorted ascending with the index", mins, maxes
+            f"Partitions must be sorted ascending by {column.name}", mins, maxes
+        )
+    return mins, maxes
+
+
+def compute_divisions(df, col=None, **kwargs) -> tuple:
+    mins, maxes = compute_mins_and_maxes(df, col=col, **kwargs)
+    has_overlap = any(mins[i] <= maxes[i - 1] for i in range(1, len(mins)))
+    if has_overlap:
+        raise ValueError(
+            f"Partitions must be sorted ascending by {'index' if col is None else col}"
```

Comment:
You may have just waited many minutes to get this error message, I wonder if we can include any more information from our hard-earned mins and maxes to say which partitions overlap, what the overlapping values are, what you can do about it, etc.?

Also, the fact that there's a `fix_overlap` method makes me wonder if we should always error in this case, or if sometimes we should return the divisions with overlap but then, if you're going to set them, require you to use `fix_overlap`?

Reply:
Yes you are definitely right. I was trying to just think about optimizing for the best-case scenario, but we should think about how to give more informative errors. I am leaning towards preferring informative errors rather than trying to resolve overlap. Especially in this method that just computes existing divisions.

Reply:
Yeah, it would just be nice if in order to fix the overlap, you didn't have to recompute the divisions again, since `mins, maxes` has all the information we need.

But I'm not sure how much this matters? As a user, when would you call `compute_current_divisions`? I'd think only when you believe the current partitions are already sorted. So if they weren't actually sorted, that would indicate that some assumptions were actually wrong and that you needed to change your code (add a `sort_values` or `set_index` or something before), not necessarily that you'd want overlap automatically fixed?

Though I also don't know how `fix_overlap` works, and how much more efficient it is.